### ISSUE TITLE
[Feature Request] Check if `templ` and `tailwindcss` are installed in the Makefile

### ISSUE THREAD

**Issue by jaswdr** [Melkeydev/go-blueprint#295]
### Tell us about your feature request

When I tried to build my first project using Blueprint I noticed that the generated Makefile is assuming that the system has `templ` and `tailwindcss` installed locally. This was not true in my case, and I believe that this could happen with more people, so I think it would be nice to add a step before the `build` to check if both are installed, and if not, redirect the user to the right place to install them.

### Disclaimer

- [X] I agree

**Comment by Ujstor** [comment]
I opened ticket #275 with the same intention :) The tmpl one is simple, but Tailwind needs a little more work. 

There is a checkOS function in program.go that can return whether we are running a Windows or Unix-like OS. Expanding the checkOS function to verify if we are on Windows, Linux, or Darwin can solve complexities with Makefile templating.

```makefile
@if command -v templ > /dev/null; then \
			templ generate; \
	else \
		read -p "Go's 'templ' is not installed on your machine. Do you want to install it? [Y/n] " choice; \
		if [ "$$choice" != "n" ] && [ "$$choice" != "N" ]; then \
			go install github.com/a-h/templ/cmd/templ@latest; \
			templ generate; \
		else \
			echo "You chose not to install templ. Exiting..."; \
			exit 1; \
		fi; \
	fi
```

**Comment by Ujstor** [comment]
Resolved in #306
